test/proxy.test.js

```diff
@@ -0,0 +1,254 @@
+
+import sinon from 'sinon';
+import { ENV_VARS, LOCAL_ENV_VARS } from 'apify-shared/consts';
+import Apify from '../build/index';
+import * as requestUtils from '../build/utils_request';
+import * as utils from '../build/utils';
+
+const { apifyClient } = utils;
+
+const groups = ['GROUP1', 'GROUP2'];
+const hostname = 'proxy.apify.com';
+const port = 8000;
+const password = 'test12345';
+const country = 'CZ';
+const sessionId = 538909250932;
+const basciOpts = {
+    groups,
+    country,
+    password,
+    hostname,
+    port,
+};
+
+describe('Apify.ProxyConfiguration', () => {
+    test('should accept all options', () => {
+        const proxyConfiguration = new Apify.ProxyConfiguration(basciOpts);
+
+        expect(proxyConfiguration).toBeInstanceOf(Apify.ProxyConfiguration);
+        expect(proxyConfiguration.groups).toBe(groups);
+        expect(proxyConfiguration.country).toBe(country);
+        expect(proxyConfiguration.password).toBe(password);
+        expect(proxyConfiguration.hostname).toBe(hostname);
+        expect(proxyConfiguration.port).toBe(port);
+    });
+
+    test('should getUrl() work', () => {
+        const proxyConfiguration = new Apify.ProxyConfiguration(basciOpts);
+
+        const proxyUrl = 'http://groups-GROUP1+GROUP2,session-538909250932,country-CZ:test12345@proxy.apify.com:8000';
+        expect(proxyConfiguration.getUrl(sessionId)).toBe(proxyUrl);
+    });
+
+    test('should getInfo() work', () => {
+        const proxyConfiguration = new Apify.ProxyConfiguration(basciOpts);
+        const url = 'http://groups-GROUP1+GROUP2,session-538909250932,country-CZ:test12345@proxy.apify.com:8000';
+
+        const proxy = {
+            sessionId,
+            url,
+            groups,
+            country,
+            password,
+            hostname,
+            port,
+        };
+        expect(proxyConfiguration.getInfo(sessionId)).toStrictEqual(proxy);
+    });
+
+    test('should throw invalid arguments structure', () => {
+        let opts;
+        // Group value
+        const invalidGroups = ['GROUP1*'];
+        opts = Object.assign({}, basciOpts);
+        opts.groups = invalidGroups;
+        try {
+            // eslint-disable-next-line no-unused-vars
+            const proxyConfiguration = new Apify.ProxyConfiguration(opts);
+            throw new Error('wrong error');
+        } catch (err) {
+            expect(err.message).not.toBe('wrong error');
```

Comment:
This is a bit overkill. If it matches the below, then it cannot be `wrong error`.

Reply:
Fixed.